zkevm-circuits/src/bytecode_circuit/bytecode_chiquito.rs

```diff
@@ -0,0 +1,217 @@
+    BytecodeCircuitConfigArgs {
+        bytecode_table,
+        keccak_table,
+        challenges,
+    }: &BytecodeCircuitConfigArgs<F>,
+    push_data_table_value: Column<Fixed>,
+    push_data_table_size: Column<Fixed>,
+) -> Circuit<F, WitnessInput<F>, RefCell<BytecodeWitnessGen<F>>> {
+    let bytecode_circuit = circuit::<F, WitnessInput<F>, RefCell<BytecodeWitnessGen<F>>, _>(
+        "bytecode circuit",
+        |ctx| {
+            use chiquito::dsl::cb::*;
+
+            let length = ctx.forward("length");
+            let push_data_left = ctx.forward("push_data_left");
+            let value_rlc = ctx.forward_with_phase("value_rlc", 1); // 1 -> SecondPhase
+
+            let index = ctx.import_halo2_advice("index", bytecode_table.index);
+            let hash = ctx.import_halo2_advice("hash", bytecode_table.code_hash);
+            let is_code = ctx.import_halo2_advice("is_code", bytecode_table.is_code);
+            let value = ctx.import_halo2_advice("value", bytecode_table.value);
+            let tag = ctx.import_halo2_advice("tag", bytecode_table.tag);
+
+            let push_data_table_value =
+                ctx.import_halo2_fixed("push_data_value", push_data_table_value);
+            let push_data_table_size =
+                ctx.import_halo2_fixed("push_data_size", push_data_table_size);
+
+            let keccak_is_enabled =
+                ctx.import_halo2_advice("keccak_is_enabled", keccak_table.is_enabled);
+            let keccak_value_rlc =
+                ctx.import_halo2_advice("keccak_value_rlc", keccak_table.input_rlc);
+            let keccak_length = ctx.import_halo2_advice("keccak_length", keccak_table.input_len);
+            let keccak_hash = ctx.import_halo2_advice("keccak_hash", keccak_table.output_rlc);
+
+            let header = ctx.step_type("header");
+            let byte_step = ctx.step_type("byte");
+
+            ctx.pragma_first_step(header);
+            ctx.pragma_last_step(header);
+
+            ctx.step_type_def(header, move |ctx| {
+                ctx.constr(isz(index));
+                ctx.constr(eq(value, length));
+
+                ctx.transition(if_next_step(header, isz(length)));
+
+                let empty_hash = rlc(
+                    &EMPTY_CODE_HASH_LE.map(|v| (v as u64).expr()),
+                    challenges.evm_word(),
+                );
+
+                ctx.transition(if_next_step(header, eq(hash, empty_hash)));
+
+                ctx.transition(if_next_step(byte_step, eq(length, length.next())));
+                ctx.transition(if_next_step(byte_step, isz(index.next())));
+                ctx.transition(if_next_step(byte_step, eq(is_code.next(), 1)));
+                ctx.transition(if_next_step(byte_step, eq(hash, hash.next())));
+                ctx.transition(if_next_step(byte_step, eq(value_rlc.next(), value.next())));
+
+                ctx.wg(move |ctx, wit| {
+                    let wit = wit.borrow();
+
+                    ctx.assign(tag, 0.field());
```

Comment:
this might not be necessary anymore, I need to check, as new step selector builder uses this column